### PR TITLE
Fixed scroll back to top button issue

### LINKED ISSUE #628: Hide "Scroll back to top" button when already on top
Hi! @srish @tuxology, Scroll to top button should not be visible when user is already at the top of the website.

image

Thanks!

### PR DESCRIPTION
**Pull request by ipriyanshi1708**
**This Pull request make the following changes:**
* Fixes issue #628  

**Checklist:**
- [x] I have created an issue for this solution I am submitting.
- [x] I created a new branch from the correct branch and made my changes in the new branch created.
- [x] My pull request is sent to the correct base branch in the original https://github.com/unstructuredstudio/zubhub repo.
- [x] I have linked the issue I worked on to this pull request submitted by me.
- [x] File(s) or Folder(s) now contain changes as specified in the issue I worked on.
- [x] I certify that I ran my checklist.

Ping @srish @tuxology @ajain1921 @NdibeRaymond 

Closes #628

### CODE REVIEW

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx [hunk 1]
```diff
@@ -142,6 +147,22 @@ function PageWrapper(props) {
     [],
   );
 
+  //Logic behind showing and hiding scroll to top button
+  const handleScroll = ()=>{
+    if(window.pageYOffset > 250){
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: How did we decide on the 250 value here?

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx [hunk 2]
```diff
@@ -89,6 +89,11 @@ function PageWrapper(props) {
   const classes = useStyles();
   const common_classes = useCommonStyles();
   const trigger = useScrollTrigger();
+
+  //Created new state "show" to show and hide the scroll to top button
+  const [show, setShow] = useState(false);
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: Remove comment and change the name to `showScrollBar`, `setShowScrollBar`. We should aim to made code more descriptive and comments should remain for only complex decisions int he code that have been taken or to describe APIs well.

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx [hunk 3]
```diff
@@ -142,6 +147,22 @@ function PageWrapper(props) {
     [],
   );
 
+  //Logic behind showing and hiding scroll to top button
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: remove comment

**Review by tuxology — CHANGES_REQUESTED** [review]
Left some comments to address.

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx (hunk 1 above)
Comment (ipriyanshi1708) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: I thought about it and tested it once again and i found it is a very large gap and button gets displayed very late. So, I think we should keep it 70, at 70 scroll back to top button will be displayed when scrollbar crosses more than half of the navbar. I am open to suggestions.

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx (hunk 2 above)
Comment (ipriyanshi1708) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: Ok I will do it.

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx (hunk 3 above)
Comment (ipriyanshi1708) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: Ok I will do it

**Comment by ipriyanshi1708** [comment]
> Left some comments to address.

Do I need to do anything else other than these 3 problems which are mentioned above?

File: zubhub_frontend/zubhub/package.json [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "axios": "^0.27.2",
     "classnames": "^2.2.6",
     "compressorjs": "^1.0.7",
+    "concurrently": "^7.6.0",
```
Comment (tuxology) [inline zubhub_frontend/zubhub/package.json]: Why have we added this? We want to avoid new dependencies being added unless really required.

Comment (ipriyanshi1708) [reply zubhub_frontend/zubhub/package.json]: Oh sorry! that's my mistake. Actually I have added that when I was trying to run both backend and frontend simultaneously on my computer. Now I have fixed it.

**Review by tuxology — APPROVED** [review]

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx [hunk 5]
```diff
@@ -142,6 +143,21 @@ function PageWrapper(props) {
     [],
   );
 
+  const handleScroll = ()=>{
+    if(window.pageYOffset > 70){
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: @kamthamc Is there a better way to calculate this? I am not sure about hardcoding value here that @ipriyanshi1708 has done.

**Comment by tuxology** [comment]
@ipriyanshi1708 Can we also use ZubHub's primary red color here instead of the default?

**Comment by ipriyanshi1708** [comment]
> @tuxology

Yes, we can surely use the primary red color instead of this color but I think this color increases the visibility of the button. I am open to discussion.

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx (hunk 5 above)
Comment (chaitanya-cloudknox) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: Why not use ref to the top element and show the button if the ref its not in viewport 

https://blog.devgenius.io/how-to-tell-if-an-html-element-is-visisble-with-javascript-9f5821056e08

Comment (tuxology) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: @ipriyanshi1708 Please make the changes proposed by @chaitanya-cloudknox above 👆

**Review by tuxology — CHANGES_REQUESTED** [review]
@ipriyanshi1708 Please make the changes requested by @chaitanya-cloudknox

File: zubhub_frontend/zubhub/src/views/PageWrapper.jsx (hunk 5 above)
Comment (ipriyanshi1708) [reply zubhub_frontend/zubhub/src/views/PageWrapper.jsx]: > @ipriyanshi1708 Please make the changes proposed by @chaitanya-cloudknox above 👆
Okay, I am working on that.

**Pull request closed without merging**